### PR TITLE
Add mass unban command.

### LINKED ISSUE #90: munban
you never know when some bot goes crazy with the bans

### PR DESCRIPTION
**Pull request by trilleplay**
This closes #90

### CODE REVIEW

File: GearBot/Cogs/Moderation.py [hunk 1]
```diff
@@ -334,6 +334,18 @@ def _can_act(action, ctx, user, check_bot=True):
...
                                          reason=reason, inf=i.id)
 
+    async def _unban(self, ctx, user, reason, confirm, days=0):
+        self.bot.data["forced_exits"].add(f"{ctx.guild.id}-{user.id}")
+        await ctx.guild.unban(user, reason=Utils.trim_message(
+            f"Moderator: {ctx.author.name}#{ctx.author.discriminator} ({ctx.author.id}) Reason: {reason}", 500))
+        Infraction.update(active=False).where((Infraction.user_id == user.id) & ((Infraction.type == "Ban") | (Infraction.type == "Tempban")) &
+                                              (Infraction.guild_id == ctx.guild.id)).execute()
+        i = InfractionUtils.add_infraction(ctx.guild.id, user.id, ctx.author.id, "Unban", reason)
+        GearbotLogging.log_to(ctx.guild.id, 'ban_log', user=Utils.clean_user(user), user_id=user.id, moderator=Utils.clean_user(ctx.author), moderator_id=ctx.author.id, reason=reason, inf=i.id)
```
Comment (BlackHoleFox) [inline GearBot/Cogs/Moderation.py]: If you could multi-line this one, it would greatly improve readability

Comment (AEnterprise) [inline GearBot/Cogs/Moderation.py]: can we maybe log this as unban instead of a ban? :P

File: GearBot/Cogs/Moderation.py [hunk 2]
```diff
@@ -334,6 +334,18 @@ def _can_act(action, ctx, user, check_bot=True):
...
+        if confirm:
+            await MessageUtils.send_to(ctx, "YES", "ban_confirmation", user=Utils.clean_user(user), user_id=user.id,
```
Comment (AEnterprise) [inline GearBot/Cogs/Moderation.py]: same here

File: GearBot/Cogs/Moderation.py [hunk 3]
```diff
@@ -374,6 +386,45 @@ def _can_act(action, ctx, user, check_bot=True):
         else:
             await self.empty_list(ctx, "ban")
 
+    @commands.guild_only()
+    @commands.command()
+    @commands.bot_has_permissions(ban_members=True, add_reactions=True)
+    async def munban(self, ctx, targets: Greedy[PotentialID], *, reason: Reason = ""):
+        """mban_help"""
```
Comment (AEnterprise) [inline GearBot/Cogs/Moderation.py]: and here

File: GearBot/Cogs/Moderation.py [hunk 4]
```diff
@@ -374,6 +386,45 @@ def _can_act(action, ctx, user, check_bot=True):
...
+            pmessage = await MessageUtils.send_to(ctx, "REFRESH", "processing")
+            valid = 0
+            failures = []
+            for t in targets:
+                try:
+                    user = await DiscordUser().convert(ctx, str(t))
+                except BadArgument as bad:
+                    failures.append(f"{t}: {bad}")
+                else:
+                    allowed, message = self._can_act("ban", ctx, user)
```
Comment (AEnterprise) [inline GearBot/Cogs/Moderation.py]: you really like banning people don't you? it's a pointless check, there is nothing to check other then "are they banned or not"

File: GearBot/Cogs/Moderation.py [hunk 5]
```diff
@@ -334,6 +334,18 @@ def _can_act(action, ctx, user, check_bot=True):
             await MessageUtils.send_to(ctx, "YES", "ban_confirmation", user=Utils.clean_user(user), user_id=user.id,
                                          reason=reason, inf=i.id)
 
+    async def _unban(self, ctx, user, reason, confirm, days=0):
+        self.bot.data["forced_exits"].add(f"{ctx.guild.id}-{user.id}")
```
Comment (AEnterprise) [inline GearBot/Cogs/Moderation.py]: this is for people that get banned, not unbanned

**Comment by trilleplay** [comment]
Alright all should hopefully be fixed, please make sure i didnt miss anything

**Pull request merged**